task8/src/main/java/PostServlets/List.java

```diff
@@ -0,0 +1,49 @@
+package PostServlets;
+import PostWork.Post;
+import PostWork.Service;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class PostList extends HttpServlet {
+
+    private Service posts = Service.getInstance();
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        int top, skip;
+        if (request.getParameter("skip") != null) {
+            skip = Integer.parseInt(request.getParameter("skip"));
+        } else {
+            skip = 0;
+        }
+        if (request.getParameter("top") != null) {
+            top = Integer.parseInt(request.getParameter("top"));
+        } else {
+            top = 0;
+        }
+        String vendor = request.getParameter("vendor");
+        String createdAt = request.getParameter("createdAt");
+
+        Map<String, String> filterConfig = new HashMap<>();
+        if (vendor != null) {
+            filterConfig.put("vendor", vendor);
+            response.getOutputStream().println("yes we have vendor" + skip + top);
```

Comment:
Does this output have any meaning? if this is an HTML page, the notice "yes we have..." will be rendered above the top of the page, stylessly. And if this is a JSON output, a line like this would just break the JSON, make it unparseable